server/api/filter.go

```diff
@@ -0,0 +1,79 @@
+package handler
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/codeday-labs/bookworms/server/db"
+	"github.com/codeday-labs/bookworms/server/utils"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+func filterReview(searchQeury []string) ([]*db.Review, error) {
```

Comment:
This function looks almost the same as the one in `server/api/reviews.go`, it should be extracted to be a utility func instead.

Reply:
They serve different purposes

Reply:
I think they share the same purpose, both of them filter data based on the input argument.

Here is one way to fix this issue:
1. Extract this func as an utility func and rename to `FilterReview(filter interface{})`
2. Use following code in filter.go

```go
FilterReview(bson.M{"categories": bson.M{"$in": searchQeury}})
```